src/components/achievements.tsx

```diff
@@ -0,0 +1,127 @@
+'use client';
+
+import React from 'react';
+import { Doughnut } from 'react-chartjs-2';
+import {
+  Chart as ChartJS,
+  ChartData,
+  Tooltip,
+  ArcElement,
+  Filler,
+  Legend,
+  Title,
+} from 'chart.js/auto';
+
+ChartJS.register(ArcElement, Tooltip, Legend, Title, Filler);
+
+interface AchievementData {
+  difficulty: number; // 難易度 (1: Easy, 2: Normal, 3: Hard)
+  status: 'submitted' | 'reviewed' | '';
+}
+
+export const Achievements: React.FC<{ data: AchievementData[] }> = ({
+  data,
+}) => {
+  // 進捗度
+  const easyCount = data.filter(
+    (item) =>
+      item.difficulty === 1 &&
+      (item.status === 'submitted' || item.status === 'reviewed')
+  ).length;
+  const normalCount = data.filter(
+    (item) =>
+      item.difficulty === 2 &&
+      (item.status === 'submitted' || item.status === 'reviewed')
+  ).length;
+  const hardCount = data.filter(
+    (item) =>
+      item.difficulty === 3 &&
+      (item.status === 'submitted' || item.status === 'reviewed')
+  ).length;
+
+  const max_easy = data.filter((item) => item.difficulty === 1).length;
+  const max_normal = data.filter((item) => item.difficulty === 2).length;
+  const max_hard = data.filter((item) => item.difficulty === 3).length;
```

Comment:
ちょっと計算量が多くなりそうなので、
ここは愚直に for ループでもいいかもです。
```suggestion
  type DifficultyNumber = Record<Submission['difficulty'], number>;
  // 進捗度
  const problemCount: DifficultyNumber = {
    1: 0,
    2: 0,
    3: 0,
  };
  const finishedProblemCount: DifficultyNumber = {
    1: 0,
    2: 0,
    3: 0,
  };

  data.forEach((item) => {
    problemCount[item.difficulty] += 1;
    if (item.status === 'submitted' || item.status === 'reviewed') {
      finishedProblemCount[item.difficulty] += 1;
    }
  });
```

Reply:
[b12b871](https://github.com/Greek-Academy/alpha-dojo-v2/pull/43/commits/b12b8718ad171aa3e1fc492140b6ded1cf0f7cb8)で対応済み